src/main/kotlin/org/jmailen/gradle/kotlinter/support/EditorConfigUtils.kt

```diff
@@ -12,20 +15,39 @@ internal fun editorConfigOverride(ktLintParams: KtLintParams): EditorConfigOverr
     val rules = ktLintParams.disabledRules
 
     return if (rules.isEmpty()) {
-        EditorConfigOverride.emptyEditorConfigOverride
+        EMPTY_EDITOR_CONFIG_OVERRIDE
     } else {
-        EditorConfigOverride.from(DefaultEditorConfigProperties.ktlintDisabledRulesProperty to rules.joinToString(separator = ","))
+        rules
+            .asSequence()
+            .map(::getDisabledRulePropertyName)
+            .map { propertyName ->
+                EditorConfigProperty(
+                    type = PropertyType(propertyName, "Rule to be disabled", IDENTITY_VALUE_PARSER),
+                    defaultValue = "disabled",
+                )
+            }
+            .map { it to "disabled" }
+            .toList()
+            .toTypedArray()
+            .let { EditorConfigOverride.from(*it) }
     }
 }
 
-internal fun resetEditorconfigCacheIfNeeded(
+private fun getDisabledRulePropertyName(it: String) =
```

Comment:
🎨 
```suggestion
private fun getDisabledRulePropertyName(ruleName: String) =
```

Reply:
\+ I can propose dropping the `disabled` suffix and clarify this method transform rule name to Ktlint EditorConfig property name
```suggestion
private fun getKtlintPropertyName(ruleName: String) =
```